ctr/main.go

```diff
@@ -50,5 +63,5 @@ func main() {
 
 func fatal(err string, code int) {
 	fmt.Fprintf(os.Stderr, "[ctr] %s\n", err)
-	os.Exit(code)
+	panic(Exit{code})
```

Comment:
`os.Exit` doesn't execute deferred routine, `panic` does.

It's caught up by the `recover` call in the defer routine  I added to `main()`.

As long as the parameter to `panic` is of type `Exit` the program will exit normally via `os.Exit` if it's not it will panic as expected.